ldi-core/ldes-client-v2/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/LdesProvider.java

```diff
@@ -0,0 +1,28 @@
+package ldes.client.treenodesupplier;
+
+import ldes.client.requestexecutor.executor.RequestExecutor;
+import ldes.client.startingtreenode.StartingTreeNodeFinder;
+import ldes.client.startingtreenode.domain.valueobjects.Endpoint;
+import ldes.client.startingtreenode.domain.valueobjects.TreeNode;
+import ldes.client.treenodesupplier.domain.valueobject.Ldes;
+import ldes.client.treenodesupplier.exception.NoStartingNodeException;
+import org.apache.jena.riot.Lang;
+
+import java.util.Optional;
+
+public class LdesProvider {
+
+	private final RequestExecutor requestExecutor;
+
+	public LdesProvider(RequestExecutor requestExecutor) {
+		this.requestExecutor = requestExecutor;
+	}
+
+	public Ldes getLdes(String url, Lang lang) {
```

Comment:
I am not a big fan of the 'Ldes' object in this setting. It is actually a wrapper of the starting-node package, can't we call it that? "StartingTreeNode" ?